src/plugins/link/main/ts/core/Utils.ts

```diff
@@ -10,8 +10,11 @@ import { Arr, Option } from '@ephox/katamari';
 import Editor from 'tinymce/core/api/Editor';
 import Tools from 'tinymce/core/api/util/Tools';
 import Settings from '../api/Settings';
+import { ForceExternalTargets } from '../api/Types';
 import { AttachState, LinkDialogOutput } from '../ui/DialogTypes';
 
+const protocolRe = /^\w+:/i;
```

Comment:
Caching regex in JS is generally bad. Please make this a function e.g. `/^\w+:/i.test(str)`

Reply:
I'm going to make the change, a function works better here. Would you mind expanding or pointing me to a resource explaining why caching regexp in JS is generally bad?

Reply:
Because regex maintains a mutable state. The simple example is using a `/g` regex:

```
$ node
Welcome to Node.js v12.3.1.
Type ".help" for more information.
> regex = /a/g
/a/g
> regex.test('a')
true
> regex.test('a')
false
> 
```

Even if there's no risk of this happening in one specific example, I prefer to have all regexes created on demand so it can't happen by accident.

Reply:
Thanks for explaining. That's indeed a tricky behavior!